src/frontend/src/lib/components/networks/NetworksTestnetsToggle.svelte

```diff
@@ -1,13 +1,27 @@
 <script lang="ts">
 	import { Toggle } from '@dfinity/gix-components';
+	import { goto } from '$app/navigation';
+	import { NETWORK_PARAM } from '$lib/constants/routes.constants';
 	import { testnetsEnabled } from '$lib/derived/settings.derived';
 	import { i18n } from '$lib/stores/i18n.store';
 	import { testnetsStore } from '$lib/stores/settings.store';
 
+	// TODO: create tests for this component once we have testId from GIX-CMP
+	// PR: https://github.com/dfinity/gix-components/pull/531
+
 	let checked: boolean;
 	$: checked = $testnetsEnabled;
 
-	const toggleTestnets = () => testnetsStore.set({ key: 'testnets', value: { enabled: !checked } });
+	const toggleTestnets = async () => {
+		// Reset network param, since the network is selectable only when testnets are enabled
+		if (checked) {
+			const url = new URL(window.location.href);
+			url.searchParams.delete(NETWORK_PARAM);
+			await goto(url, { replaceState: true, noScroll: true });
```

Comment:
I ask myself if we should use `goto` or imperatively remove the param for the URL because, does `goto` implies that some stores will be reevaluted including `testnetsStore` twice with next line?

Reply:
i think that doing it imperatively will not "refresh" some data of the `page` store; BUT, I am not sure we need the `page` store in our cases; I just wanted to give a clean "refresh", let's say

Reply:
Yeah but that's the thing, a "clean refresh" is a really different behavior than the current one, which is just updating the testnet store that's why to be honest I lean to "just" updating the URL here.

Reply:
Discussed offline, I did not understood. `goto` is a must here as we really want to update the store since we remove the network information.

Reply:
As discussed offline, we will got with goto because we actually need to trigger a refresh of the `page` store of svelte, even if it costs to refresh derived that we don't need